server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -2112,11 +2112,13 @@ private boolean doCancelMaintenance(final long hostId) {
 
         /* TODO: move to listener */
         _haMgr.cancelScheduledMigrations(host);
+
+        boolean vms_migrating = false;
         final List<VMInstanceVO> vms = _haMgr.findTakenMigrationWork();
         for (final VMInstanceVO vm : vms) {
             if (vm != null && vm.getHostId() != null && vm.getHostId() == hostId) {
                 s_logger.info("Unable to cancel migration because the vm is being migrated: " + vm);
```

Comment:
@alexandrelimassantana 
The reason for not breaking from the for loop is to log info about vms that are under migration. Probably the log level should be increased to warn. These log messages would be valuable for trouble shooting.
On readability front yes the code will be improved further.